geode-core/src/test/java/org/apache/geode/internal/offheap/ReferenceCountHelperImplTest.java

```diff
@@ -12,976 +12,966 @@
-    rchi = getTrueTrue();
+    ReferenceCountHelperImpl referenceCountHelperImpl = newReferenceCountHelperImpl_TrueTrue();
     String owner = null;
-    rchi.setReferenceCountOwner(owner);
-    AtomicInteger ai = rchi.getReenterCount();
-    assertEquals(0, ai.get());
-
-    owner = new String("SomeOwner");
-    rchi.setReferenceCountOwner(owner);
-    ai = rchi.getReenterCount();
-    assertEquals(1, ai.get());
-    assertEquals(rchi.getReferenceCountOwner(), owner);
-
-    String owner2 = new String("SomeOwner2");
-    rchi.setReferenceCountOwner(owner2);
-    ai = rchi.getReenterCount();
-    assertEquals(2, ai.get());
-    assertTrue(rchi.getReferenceCountOwner() != owner2); // stays original owner until cnt = 0
+    referenceCountHelperImpl.setReferenceCountOwner(owner);
+    AtomicInteger reenterCount = referenceCountHelperImpl.getReenterCount();
+    assertThat(reenterCount.get()).isZero();
+
+    owner = "SomeOwner";
+    referenceCountHelperImpl.setReferenceCountOwner(owner);
+    reenterCount = referenceCountHelperImpl.getReenterCount();
+    assertThat(reenterCount.get()).isOne();
+    assertThat(referenceCountHelperImpl.getReferenceCountOwner()).isEqualTo(owner);
+
+    String owner2 = "SomeOwner2";
+    referenceCountHelperImpl.setReferenceCountOwner(owner2);
+    reenterCount = referenceCountHelperImpl.getReenterCount();
+    assertThat(reenterCount.get()).isEqualTo(2);
+    // stays original owner until cnt = 0
+    assertThat(referenceCountHelperImpl.getReferenceCountOwner()).isNotEqualTo(owner2);
 
     String owner3 = null;
-    rchi.setReferenceCountOwner(owner3);
-    ai = rchi.getReenterCount();
-    assertEquals(1, ai.get());
-    assertEquals(rchi.getReferenceCountOwner(), owner);
+    referenceCountHelperImpl.setReferenceCountOwner(owner3);
+    reenterCount = referenceCountHelperImpl.getReenterCount();
+    assertThat(reenterCount.get()).isOne();
+    assertThat(referenceCountHelperImpl.getReferenceCountOwner()).isEqualTo(owner);
 
     owner = null;
-    rchi.setReferenceCountOwner(owner);
-    ai = rchi.getReenterCount();
-    assertEquals(0, ai.get());
-    assertEquals(rchi.getReferenceCountOwner(), null);
+    referenceCountHelperImpl.setReferenceCountOwner(owner);
+    reenterCount = referenceCountHelperImpl.getReenterCount();
+    assertThat(reenterCount.get()).isZero();
+    assertThat(referenceCountHelperImpl.getReferenceCountOwner()).isNull();
 
-    RegionEntry re = mock(RegionEntry.class);
-    rchi.setReferenceCountOwner(re);
-    ai = rchi.getReenterCount();
-    assertEquals(1, ai.get());
-    assertEquals(rchi.getReferenceCountOwner(), re);
+    RegionEntry regionEntry = mock(RegionEntry.class);
+    referenceCountHelperImpl.setReferenceCountOwner(regionEntry);
+    reenterCount = referenceCountHelperImpl.getReenterCount();
+    assertThat(reenterCount.get()).isOne();
+    assertThat(regionEntry).isEqualTo(referenceCountHelperImpl.getReferenceCountOwner());
 
     Long address = (long) 0x1000;
```

Comment:
Multiple compiler warnings in this class can be fixed by replacing lines like this with just `long address = 0x1000;`. Alternately, as the address used is always the same in every test, it could be extracted to a constant.

Reply:
It's too difficult to explain why I left it as `Long` so I'll just make the change. This change will cause the test to take slightly longer to run because it will change a `long` back to `Long` for use as a key in a map.